addons/hr_timesheet/migrations/10.0.1.0/post-migration.py

```diff
@@ -0,0 +1,50 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 Tecnativa - Pedro M. Baeza
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from openupgradelib import openupgrade
+
+
+def migrate_allow_timesheets(env):
+    openupgrade.logged_query(
+        env.cr,
+        """
+        UPDATE project_project pp
+        SET allow_timesheets = True
+        FROM account_analytic_account aaa
+        WHERE aaa.id = pp.analytic_account_id
+        AND aaa.%s
+        """ % openupgrade.get_legacy_name('use_timesheets'),
+    )
+
+
+def fill_analytic_line_project(env):
+    """Fill project with the linked one in the related task or issue."""
```

Comment:
The migration with project_issue should occur as part of the migration of project_issue module. If a database does not have the column issue_id because that module does not exist, this will break the migration here.

Same with project_task. Does not belong here.

Also, in v9 the account_analytic_line does not have a field task_id defined in hr_timesheet module.